springwolf-plugins/springwolf-cloud-stream-plugin/src/main/java/io/github/springwolf/plugins/cloudstream/asyncapi/scanners/common/FunctionalChannelBeanBuilder.java

```diff
@@ -89,8 +90,11 @@ private static String getElementName(AnnotatedElement element) {
 
     private List<Class<?>> getTypeGenerics(AnnotatedElement element) {
         if (element instanceof Method m) {
-            ParameterizedType genericReturnType = (ParameterizedType) m.getGenericReturnType();
-            return getTypeGenerics(genericReturnType);
+            if (m.getGenericReturnType() instanceof ParameterizedType) {
```

Comment:
I did not add the type check because we are always dealing here with methods that return `Consumer`, `Supplier` or `Function` for which `getGenericReturnType` is always `ParameterizedType`. But that's probably good practice.
Maybe use smart casting though?
```java
if (m.getGenericReturnType() instanceof ParameterizedType parameterizedType) {
    return getTypeGenerics(parameterizedType);
}
```

Reply:
I did run into an exception, but could have been an intermediate state.